src/order/rel_classes.lean

```diff
@@ -303,3 +273,173 @@ instance is_trans_preimage_snd {r : α → α → Prop} [h : is_trans α r] :
   is_trans (α × α) (prod.snd ⁻¹'o r) := ⟨λ _ _ _, trans_of r⟩
 
 end prod
+
+/-! ### Strict-non strict relations -/
+
+/-- An unbundled relation class stating that `r` is the nonstrict relation corresponding to the
+strict relation `s`. Compare `preorder.lt_iff_le_not_le`. This is mostly meant to provide dot
+notation on `(⊆)` and `(⊂)`. -/
+class is_nonstrict_strict_order (α : Type*) (r s : α → α → Prop) :=
```

Comment:
Do you think `s` should be an out-param here, since it's uniquely determined in terms of `r`? @Vierkantor

Reply:
Ideally, we'd indicate that synthesis should only be blocked in the "ready to synthesize" check when both `r` and `s` are missing, since `s` and `r` uniquely determine each other. But we can't quite express that in Lean at the moment. Making one of them an `out_param` forces that one to always be a free variable during synthesis, so it would break cases where `r` is inferred from `s`. I think in the end, not having either of them as `out_param` would be the best.

Reply:
The reason for leaving `α` explicit instead of:
```lean
class is_nonstrict_strict_order {α : Type*} (r s : α → α → Prop) :=
```
is that you can write `is_nonstrict_strict_order ℕ (≤) (<)` instead of having to type-ascript the relations, right?

Reply:
`s` does **not** uniquely determine `r`. Else, we wouldn't the distinction between preorders and partial orders.

Reply:
Ah right, sorry. In any case, there is already code where it does, like `is_nonstrict_strict_order.to_is_irrefl`, so making `s` an `out_param` would still not work.